server/aesdsocket.c

```diff
@@ -108,70 +226,68 @@ printf("v2 \n");
-        if (file_fd < 0) {
-            syslog(LOG_ERR, "File not opened");
-            close(client_fd);
+    
+    while (Exit_Flag==0)
+    {
+    	 client_File_ptr = malloc(sizeof(int));  // Allocate memory 
+        if (client_File_ptr==0) {
+            syslog(LOG_ERR, "Memory Not Allocated");
             continue;
         }
-        else {
-       syslog(LOG_INFO, "File opened");
-     }
-
-        // Receive Data
-        ssize_t bytes_rec;
-        while ((bytes_rec = recv(client_fd, Str_buffer, sizeof(Str_buffer) - 1, 0)) > 0) {
+        else{
+        	syslog(LOG_INFO, "Memory Allocated");
+        }
         
-            Str_buffer[bytes_rec] = '\0';  // Null terminate
-            
-            if (write(file_fd, Str_buffer, bytes_rec) != bytes_rec) {
-                syslog(LOG_ERR, "Write operation failed.");
-            }
-             else {
-	       syslog(LOG_INFO, "Write operation Sucess");
-	     }
-            
-            fsync(file_fd);  // Flush disk
-
-            
-            if (strchr(Str_buffer, '\n')) {// if newline detected
-                close(file_fd);
-                file_fd = open(FILE_PATH, O_RDONLY);
-                if (file_fd < 0) {
-                    syslog(LOG_ERR, "reopening file Error");
-                    break;
-                }
-                 else {
-	       syslog(LOG_INFO, "reopening File Sucess");
-	     }
-
-                memset(Str_buffer, 0, sizeof(Str_buffer));
-                
-                for (; (bytes_rec = read(file_fd, Str_buffer, sizeof(Str_buffer))) > 0;) {
-                    send(client_fd, Str_buffer, bytes_rec, 0);
-                }
-
-                close(file_fd);
-                break;
-            }
+        *client_File_ptr = accept(server_fd, (struct sockaddr*)&client_addr, &ClientLen);
+        
+        if (*client_File_ptr < 0) {
+            free(client_File_ptr);
+            continue;
         }
-
-        close(file_fd);
         
-        syslog(LOG_INFO, "Closed connection from %s", inet_ntoa(client_addr.sin_addr));
+        syslog(LOG_INFO, "Accepted connection from %s", inet_ntoa(client_addr.sin_addr));
+        
+        thread_node_t *NewNode = malloc(sizeof(thread_node_t)); //Create New Node for Handling CLient
+        if(NewNode==0)
+        {
+           syslog(LOG_ERR, "Memory Not allocated for thread ");
+           close(*client_File_ptr);
+           free(client_File_ptr);
+           continue;
+        }
+        else
+        {
+          syslog(LOG_INFO, "Memory Allocated for Thread");
+        }
+        
+        NewNode->client_fd = *client_File_ptr;
+        pthread_create(&NewNode->thread, NULL, ClientHandle, client_File_ptr);
+        SLIST_INSERT_HEAD(&HeadNode, NewNode, NextNode);
         
-        close(client_fd);
     }
+    
+    thread_node_t *node;
+    
+    while (SLIST_EMPTY(&HeadNode)==0) { //wait till all threads finish
+        node = SLIST_FIRST(&HeadNode);
+        pthread_join(node->thread, NULL);
```

Comment:
What is a potential issue with only `pthread_join` ing outside the `while (Exit_Flag==0)` loop?